src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -363,23 +382,24 @@ public Study convertDatasetToStudy(User user, Dataset dataset, StudyConversion s
   }
 
   public Study updateStudyCustodians(User user, Integer studyId, String custodians) {
-    logInfo(String.format("User %s is updating custodians for study id: %s; custodians: %s", user.getEmail(), studyId, custodians));
+    logInfo(String.format("User %s is updating custodians for study id: %s; custodians: %s",
+        user.getEmail(), studyId, custodians));
     Study study = studyDAO.findStudyById(studyId);
     if (study == null) {
       throw new NotFoundException("Study not found");
     }
-    Optional<StudyProperty> optionalProp = study.getProperties() == null ?
-        Optional.empty() :
-        study
-        .getProperties()
-        .stream()
-        .filter(p -> p.getKey().equals(dataCustodianEmail))
-        .findFirst();
-    if (optionalProp.isPresent()) {
-      studyDAO.updateStudyProperty(studyId, dataCustodianEmail, PropertyType.Json.toString(), custodians);
+    boolean propPresent = study.getProperties().stream()
+        .anyMatch(prop -> prop.getKey().equals(dataCustodianEmail));
+    if (propPresent) {
+      studyDAO.updateStudyProperty(studyId, dataCustodianEmail, PropertyType.Json.toString(),
+          custodians);
     } else {
-      studyDAO.insertStudyProperty(studyId, dataCustodianEmail, PropertyType.Json.toString(), custodians);
+      studyDAO.insertStudyProperty(studyId, dataCustodianEmail, PropertyType.Json.toString(),
+          custodians);
     }
+    List<Dataset> datasets = study.getDatasetIds().isEmpty() ? List.of()
+        : datasetDAO.findDatasetsByIdList(study.getDatasetIds());
```

Comment:
Forcing this check into the caller will result in duplicate code. I believe JDBI will correctly handle empty lists, but if there's a concern, I would add a wrapper function to the DAO API so callers don't have to worry about this.

Reply:
I'm not that familiar with JDBI but it looks like the behavior is configurable. https://jdbi.org/apidocs/org/jdbi/v3/sqlobject/customizer/BindList.EmptyHandling.html.